components/supervisor/pkg/supervisor/supervisor.go

```diff
@@ -318,7 +318,12 @@ func Run(options ...RunOption) {
 			log.Debugf("unshallow of local repository took %v", time.Since(start))
 		}()
 
-		cmd := runAsGitpodUser(exec.Command("git", "fetch", "--depth", "20"))
+		var cmd *exec.Cmd
+		if cfg.isHeadless() {
+			cmd = runAsGitpodUser(exec.Command("git", "fetch", "--depth", "20"))
+		} else {
+			cmd = runAsGitpodUser(exec.Command("git", "fetch", "--unshallow", "--tags"))
```

Comment:
If we do this for all workspaces, chances are we'll be consuming considerable CPU time again at the cost of workspace startup time.

It might be useful as a stop-gap measure though.